ReactWindows/ReactNative/Views/Web/ReactWebViewManager.cs

```diff
@@ -0,0 +1,306 @@
+        private const int CommandReload = 3;
+
+        private readonly Dictionary<int, string> _injectedJS = new Dictionary<int, string>();
+
+        /// <summary>
+        /// The name of the view manager.
+        /// </summary>
+        public override string Name
+        {
+            get
+            {
+                return "RCTWebView";
+            }
+        }
+
+        /// <summary>
+        /// The commands map for the webview manager.
+        /// </summary>
+        public override IReadOnlyDictionary<string, object> CommandsMap
+        {
+            get
+            {
+                return new Dictionary<string, object>
+                {
+                    { "goBack", CommandGoBack },
+                    { "goForward", CommandGoForward },
+                    { "reload", CommandReload },
+                };
+            }
+        }
+
+        /// <summary>
+        /// Sets whether JavaScript is enabled or not.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="enabled"></param>
+        [ReactProp("javaScriptEnabled")]
+        public void SetJavaScriptEnabled(WebView view, bool enabled)
+        {
+            view.Settings.IsJavaScriptEnabled = enabled;
+        }
+
+        /// <summary>
+        /// CSets whether Indexed DB is enabled or not.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="enabled"></param>
+        [ReactProp("indexedDbEnabled")]
+        public void SetIndexedDbEnabled(WebView view, bool enabled)
+        {
+            view.Settings.IsIndexedDBEnabled = enabled;
+        }
+
+        /// <summary>
+        /// Sets the JS to be injected when the webpage loads.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="injectedJavaScript"></param>
+        [ReactProp("injectedJavaScript")]
+        public void SetInjectedJavaScript(WebView view, string injectedJavaScript)
+        {
+            if (_injectedJS.ContainsKey(view.GetTag()))
+            {
+                _injectedJS[view.GetTag()] = injectedJavaScript;
+            }
+            else
+            {
+                _injectedJS.Add(view.GetTag(), injectedJavaScript);
+            }           
+        }
+
+        /// <summary>
+        /// Sets webview source.
+        /// </summary>
+        /// <param name="view">a webview instance.</param>
+        /// <param name="source"></param>
+        [ReactProp("source")]
+        public void SetSource(WebView view, JObject source)
+        {
+            if (source != null)
+            {
+                var html = source.Value<string>("html");
+                if (html != null)
+                {
+                    var baseUrl = source.Value<string>("baseUrl");
+                    if (baseUrl != null)
+                    {
+                        view.Source = new Uri(baseUrl);
```

Comment:
I'd say so for now.  Input validation should be done in JavaScript anyway (where there is more control over how warnings / errors are handled)

---
In reply to: [60720808](https://github.com/ReactWindows/react-native/pull/372#discussion_r60720808) [](ancestors = 60720808)